internal/document/identity/resolver.go

```diff
@@ -79,9 +96,16 @@ func (ir *IdentityResolver) GetCellID(obj interface{}, attributes map[string]str
 	id := ulid.GenerateID()
 	ir.cache[obj] = id
 
+	// TODO(cepeda): I'm not entirely confident that this is the optimal approach.
```

Comment:
@pastuxso let's spin this out in a separate ticket, please. I've refactored this code under time pressure and want to make sure we're not overlooking anything.